source/scripts/Background/controllers/MainController.ts

```diff
@@ -300,6 +300,19 @@ const MainController = (walletState): IMainController => {
           wallet,
         })
       );
+      const isTestnet = verifyIfIsTestnet();
+      const cannotContinueWithTrezorAccount =
+        (isTestnet && activeAccount.type === KeyringAccountType.Trezor) ||
+        (!isBitcoinBased && activeAccount.type === KeyringAccountType.Trezor) ||
+        accounts[activeAccount.type][activeAccount.id]?.originNetwork !==
+          network;
+
+      if (cannotContinueWithTrezorAccount) {
```

Comment:
If the user is connected on Trezor and with it as an active account and switches to a testnet we should display a warning to him and display all accounts so he can choose one to use